xarray/tests/test_variable.py

```diff
@@ -785,36 +785,69 @@ def test_getitem_error(self):
         with raises_regex(IndexError, "Dimensions of indexers mis"):
             v[:, ind]
 
-    def test_pad(self):
+    @pytest.mark.parametrize(
+        "mode",
+        [
+            "mean",
+            pytest.param("median", marks=pytest.mark.xfail),
```

Comment:
let's do
```
            pytest.param("median", marks=pytest.mark.xfail(reason="not supported by dask")),
            pytest.param("reflect", marks=pytest.mark.xfail(reason="dask.array.pad bug")),
```

Reply:
I totally agree that it looks weird using the `np.nan`, but that is the way numpy ints handle NaN's. `xr.shift` has  the same problem
```
>>> arr = xr.DataArray([1,2,3])
>>> arr.shift(dim_0=1, fill_value=np.nan)
<xarray.DataArray (dim_0: 3)>
array([-2147483648,           1,           2])
Dimensions without coordinates: dim_0
```

The xfails with the reason looks much better I agree :-)

Reply:
right that's how nan is represented with int. My point was that the output was cast to float in one case and not the other; and it's not very obvious why. Sorry, I should've been clearer.

Reply:
It's the difference between `xarray.dtypes.NA` and `np.nan`.